### PR TITLE
Combine Windows and Linux CI Workflows

### PR DESCRIPTION
**Pull request by mic-max**
Towards https://github.com/open-telemetry/opentelemetry-dotnet/issues/3014

## Changes

Merged the Windows and Linux specific build test CI workflows into a single one.

- Brother PR https://github.com/open-telemetry/opentelemetry-dotnet-contrib/pull/735

Verified the markdown only changes will continue to not kickoff `dotnet restore, build, and test` steps. Seen here:
https://github.com/mic-max/opentelemetry-dotnet/pull/10 by clicking `Show all checks`

### CODE REVIEW

**Comment by mic-max** [comment]
Oops meant to submit this PR to my fork 😞😖😓

**Comment by utpilla** [comment]
This wasn't meant to be closed.

File: .github/workflows/ci-md.yml [hunk 1]
```diff
@@ -15,11 +15,16 @@ on:
...
-        version: [net6.0]
+        os: [ windows-latest, ubuntu-latest ]
+        version: [ net462, net6.0, net7.0 ]
+        exclude:
+        - os: ubuntu-latest
+          version: net462
+        - os: ubuntu-latest
+          version: net7.0
 
+    runs-on: ubuntu-latest
```
Comment (TimothyMothra) [inline .github/workflows/ci-md.yml:28]: should use the matrix here

Comment (TimothyMothra) [inline .github/workflows/ci-md.yml:28]:
```suggestion
    runs-on: ${{ matrix.os }}
```

should use the matrix here

Comment (mic-max) [reply .github/workflows/ci-md.yml:28]: Should I?
The old `.github/workflows/windows-ci-md.yml` used `ubuntu-latest`, It's just to run the `echo` command.

Comment (TimothyMothra) [reply .github/workflows/ci-md.yml:28]: When you set the Required rules in github, it's going to be based on a string containing all the values in the matrix.

For example:
- Build, windows-latest, net462
- Build, ubuntu-latest, net6

The -md workflow will need to generate those same names to match the required rules.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -12,12 +12,17 @@ on:
...
       matrix:
-        version: [net462,net6.0,net7.0]
-
+        os: [ windows-latest, ubuntu-latest ]
+        version: [ net462, net6.0, net7.0 ]
+        exclude:
+        - os: ubuntu-latest
+          version: net462
+        - os: ubuntu-latest
+          version: net7.0
```
Comment (TimothyMothra) [inline .github/workflows/ci.yml:23]: I think you can remove this exclude.
Looks like the old Linux build was running net7

Comment (mic-max) [reply .github/workflows/ci.yml]: Yes, thanks!

**Review by TimothyMothra — APPROVED** [review]

File: .github/workflows/ci-md.yml [hunk 3]
```diff
@@ -15,11 +15,14 @@ on:
...
-    runs-on: ubuntu-latest
-
     strategy:
       matrix:
-        version: [net6.0]
+        os: [ windows-latest, ubuntu-latest ]
+        version: [ net462, net6.0, net7.0 ]
+        exclude:
+        - os: ubuntu-latest
+          version: net462
```
Comment (alanwest) [inline .github/workflows/ci-md.yml:24]: Just curious... is it important that the matrix for the `md` workflow is identical to the actual CI workflow? Could all of this just be removed?

Comment (cijothomas) [reply .github/workflows/ci-md.yml:24]: it should be identical, to let CI checks show up as green for those PRs touching only md files.

**Review by alanwest — APPROVED** [review]

File: .github/workflows/ci-md.yml (hunk 3 above)
Comment (mic-max) [reply .github/workflows/ci-md.yml:24]: I actually just tried removing the strategy from the workflow on my fork and submitted a PR only modifying a `.md` file and the build workflows were skipped. I'll submit a PR to remove them from the `*-md.yml` file now.

**Pull request merged**